packages/host/app/components/ai-assistant/rename-session.gts

```diff
@@ -0,0 +1,137 @@
+    onClose: () => void;
+  };
+}
+
+export default class RenameSession extends Component<Signature> {
+  <template>
+    <AiAssistantPanelPopover data-test-rename-session ...attributes>
+      <:header>Rename Session</:header>
+      <:body>
+        <div class='rename-field'>
+          <FieldContainer @label='Session Name' @tag='label' @vertical={{true}}>
+            <BoxelInput
+              @state={{this.roomNameInputState}}
+              @value={{this.newRoomName}}
+              @errorMessage={{this.roomNameError}}
+              @onInput={{this.setName}}
+              data-test-name-field
+            />
+          </FieldContainer>
+        </div>
+        <footer class='footer'>
+          <Button
+            @kind='secondary'
+            @size='small'
+            {{on 'click' @onClose}}
+            {{onKeyMod 'Escape' @onClose}}
+            data-test-cancel-name-button
+          >
+            Cancel
+          </Button>
+          <Button
+            @kind='primary'
+            @size='small'
+            @disabled={{this.isSaveRenameDisabled}}
+            @loading={{this.doRenameRoom.isRunning}}
+            {{on 'click' this.renameRoom}}
+            {{onKeyMod 'Enter' this.renameRoom}}
+            data-test-save-name-button
+          >
+            Save
+          </Button>
+        </footer>
+      </:body>
+    </AiAssistantPanelPopover>
+
+    <style>
+      .rename-field {
+        padding: 0 var(--boxel-sp);
+      }
+      .rename-field :deep(.label) {
+        font: 700 var(--boxel-font-sm);
+      }
+      .footer {
+        display: flex;
+        justify-content: flex-end;
+        gap: var(--boxel-sp-xs);
+        padding: var(--boxel-sp);
+      }
+      .footer :deep(.boxel-button:not(:disabled)) {
+        --boxel-button-text-color: var(--boxel-dark);
+      }
+    </style>
+  </template>
+
+  @service private declare matrixService: MatrixService;
+  @tracked private newRoomName = this.args.room.name;
+  @tracked private roomNameError: string | undefined = undefined;
+
+  private get roomNameInputState() {
+    return this.roomNameError ? 'invalid' : 'initial';
+  }
+
+  private get isSaveRenameDisabled() {
+    return (
+      !this.newRoomName?.length ||
+      this.newRoomName === this.args.room.name ||
+      this.doRenameRoom.isRunning
+    );
+  }
+
+  @action
+  private setName(name: string) {
+    this.roomNameError = undefined;
+    this.newRoomName = name;
+  }
+
+  @action private renameRoom() {
+    this.doRenameRoom.perform();
+  }
+
+  private doRenameRoom = restartableTask(async () => {
+    if (!this.newRoomName.length || !this.args.room.roomId) {
+      throw new Error(`bug: should never get here`);
+    }
+    try {
+      await this.matrixService.client.setRoomName(
+        this.args.room.roomId,
+        this.newRoomName,
+      );
+      this.roomNameError = undefined;
+      this.newRoomName = '';
+      this.args.onClose();
+    } catch (e) {
+      if (isMatrixError(e)) {
```

Comment:
Maybe an error should render even if it’s not from Matrix? Like if you’re offline or there’s a DNS problem